services/data/postgres_async_db.py

```diff
@@ -286,10 +291,28 @@ async def create_record(self, record_dict):
                 cur.close()
             return DBResponse(response_code=200, body=response_body)
         except (Exception, psycopg2.DatabaseError) as error:
-            self.db.logger.exception("Exception occured")
+            self.db.logger.exception("Exception occurred")
+            return aiopg_exception_handling(error)
+
+    async def run_in_transaction_with_serializable_isolation_level(self, fun):
+        try:
+            with (
+                    await self.db.pool.cursor(
+                        cursor_factory=psycopg2.extras.DictCursor,
+                    )
+            ) as cur:
+                async with cur.begin():
```

Comment:
This is the one cursor / DB transaction wrapping most of our mutation operation.